sdk/trace/provider.go

```diff
@@ -148,23 +150,39 @@ func (p *TracerProvider) Tracer(name string, opts ...trace.TracerOption) trace.T
 		Version:   c.InstrumentationVersion(),
 		SchemaURL: c.SchemaURL(),
 	}
-	t, ok := p.namedTracer[is]
-	if !ok {
-		t = &tracer{
-			provider:             p,
-			instrumentationScope: is,
+
+	t, ok := func() (trace.Tracer, bool) {
+		p.mu.Lock()
+		defer p.mu.Unlock()
+		// Must check the flag after acquiring the mutex to avoid returning a valid tracer if Shutdown() ran
+		// after the first check above but before we acquired the mutex.
+		if p.isShutdown.Load() {
+			return trace.NewNoopTracerProvider().Tracer(name, opts...), true
+		}
+		t, ok := p.namedTracer[is]
+		if !ok {
+			t = &tracer{
+				provider:             p,
+				instrumentationScope: is,
+			}
+			p.namedTracer[is] = t
 		}
-		p.namedTracer[is] = t
-		global.Info("Tracer created", "name", name, "version", c.InstrumentationVersion(), "schemaURL", c.SchemaURL())
+		return t, ok
+	}()
+	if !ok {
+		global.Info("Tracer created", "name", name, "version", is.Version, "schemaURL", is.SchemaURL)
 	}
```

Comment:
This is like that on purpose - to not hold the lock while calling third party logging code. That code may do slow things like I/O, which would prolong the duration the lock is held, slowing down all tracing consumers. A little bit of ugliness is ok in a widely-used library to get every user a bit of performance IMO.

Another potential reason is the same why this PR exists in the first place - logging code may be instrumented with tracing and deadlock because it could try acquiring the same mutex.

Reply:
I'll add this as a comment.